### ISSUE TITLE
Support ALTER KEYSPACE in CQL

### ISSUE THREAD

**Issue by schoudhury** [YugaByte/yugabyte-db#41]
Using express-cassandra to connect to our Cassandra service for a new sample app. However, faced this error.
```
$ node server.js 
WARN: KEYSPACE ALTERED! Run the `nodetool repair` command on each affected node.
Unhandled rejection Error: SQL error (yb/yql/cql/ql/ptree/process_context.cc:52): Invalid SQL Statement. syntax error, unexpected KEYSPACE, expecting LANGUAGE
ALTER KEYSPACE "mykeyspace" WITH REPLICATION = {'class': 'SimpleStrategy','replication_factor': '1'};
      ^^^^^^^^
 (error -11)
    at ResponseError.DriverError (/Users/schoudhury/dev/express-cassandra/node_modules/cassandra-driver/lib/errors.js:14:19)
    at new ResponseError (/Users/schoudhury/dev/express-cassandra/node_modules/cassandra-driver/lib/errors.js:51:24)
    at FrameReader.readError (/Users/schoudhury/dev/express-cassandra/node_modules/cassandra-driver/lib/readers.js:317:13)
    at Parser.parseBody (/Users/schoudhury/dev/express-cassandra/node_modules/cassandra-driver/lib/streams.js:194:66)
    at Parser._transform (/Users/schoudhury/dev/express-cassandra/node_modules/cassandra-driver/lib/streams.js:137:10)
    at Parser.Transform._read (_stream_transform.js:186:10)
    at Parser.Transform._write (_stream_transform.js:174:12)
    at doWrite (_stream_writable.js:387:12)
    at writeOrBuffer (_stream_writable.js:373:5)
    at Parser.Writable.write (_stream_writable.js:290:11)
    at Protocol.ondata (_stream_readable.js:639:20)
    at emitOne (events.js:116:13)
    at Protocol.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at Protocol.Readable.push (_stream_readable.js:208:10)
```

**Comment by Gapmeister66** [comment]
I am facing the same error when using the Cassandra Java Driver version 2.1.7. I have set up my yugabyte DB as described at https://docs.yugabyte.com/latest/quick-start/install/

Please note that that I am dropping in Yugabyte to an existing application that is fully working in production with a Cassandra Database

Are there any workarounds available, or have I configured something incorrectly?

```
com.datastax.driver.core.exceptions.SyntaxError: SQL error: Invalid SQL Statement. syntax error, unexpected KEYSPACE, expecting LANGUAGE
ALTER KEYSPACE my_keyspace WITH REPLICATION = { 'class': 'SimpleStrategy',                         'replication_factor': 3 };
      ^^^^^^^^
 (error -11)
        at com.datastax.driver.core.Responses$Error.asException(Responses.java:100)
        at com.datastax.driver.core.DefaultResultSetFuture.onSet(DefaultResultSetFuture.java:118)
        at com.datastax.driver.core.RequestHandler.setFinalResult(RequestHandler.java:183)
        at com.datastax.driver.core.RequestHandler.access$2300(RequestHandler.java:45)
        at com.datastax.driver.core.RequestHandler$SpeculativeExecution.setFinalResult(RequestHandler.java:748)
        at com.datastax.driver.core.RequestHandler$SpeculativeExecution.onSet(RequestHandler.java:573)
        at com.datastax.driver.core.Connection$Dispatcher.channelRead0(Connection.java:991)
        at com.datastax.driver.core.Connection$Dispatcher.channelRead0(Connection.java:913)
        at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:105)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:339)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:324)
        at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:254)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:339)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:324)
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:339)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:324)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:242)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:339)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:324)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:847)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
        at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:111)
        at java.lang.Thread.run(Thread.java:748)

```

**Comment by kmuthukk** [comment]
hi @Gapmeister66 

1) This is the list of Cassandra statements we support currently.

https://docs.yugabyte.com/latest/api/cassandra/

2) We don't support the ALTER KEYSPACE statement yet.

3) Also, at CREATE KEYSPACE time, although we parse (don't error) if the WITH REPLICATION clause is present, YugaByte currently doesn't do anything with those. We have documented this here:

https://docs.yugabyte.com/latest/api/cassandra/ddl_create_keyspace/#semantics

So, with YugaByte DB,
```
  CREATE KEYSPACE example;
```
is sufficient, but we support this form also:
```
  CREATE KEYSPACE example WITH REPLICATION = {'class': 'SimpleStrategy', 'replication_factor': '3'}
```
but note that YugaByte DB ignores the WITH REPLICATION clause at this point. 

In other words, currently all keyspaces & tables respect the cluster level replication and placement properties. If cluster is created with RF=3, then all keyspaces will use RF=3 (for now). We don't really support keyspace level overrides yet.

That's why the ALTER TABLE keyspace also hasn't been implemented.

Are you OK with setting these properties at the cluster level, and commenting out the ALTER TABLE keyspace step from your application code when trying YugaByte DB?

**Comment by Gapmeister66** [comment]
I think that would be okay. We do also run our application off-cloud, so we would need to introduce some kind of switching behaviour. 

Would you consider ignoring the WITH_REPLICATION clause until such time that you implement it, rather than generating an error which breaks compatibility with Cassandra?

**Comment by OlegLoginov** [comment]
The diff (D5688) is on review.

**Comment by kmuthukk** [comment]
@Gapmeister66 

This support was added  in https://github.com/YugaByte/yugabyte-db/commit/8d8b55335d183a83847eb593bf5a52a11b5ec440

**Comment by OlegLoginov** [comment]
Diff D5688 was landed.
